Mathlib/Data/Set/Basic.lean

```diff
@@ -1521,6 +1521,36 @@ theorem eq_empty_of_ssubset_singleton {s : Set α} {x : α} (hs : s ⊂ {x}) : s
   ssubset_singleton_iff.1 hs
 #align set.eq_empty_of_ssubset_singleton Set.eq_empty_of_ssubset_singleton
 
+theorem Nonempty.eq_singleton_of_nonempty [Unique α] {s : Set α} (h : s.Nonempty) :
+    s = {default} := h.subset_singleton_iff.mp (fun x _ => by simp [Unique.uniq _ x])
+
+theorem Nonempty.eq_singleton_zero_of_nonempty [Zero α] [Unique α] {s : Set α} (h : s.Nonempty) :
+    s = {0} := by convert h.eq_singleton_of_nonempty
+
+theorem Nonempty.eq_singleton_one_of_nonempty [One α] [Unique α] {s : Set α} (h : s.Nonempty) :
```

Comment:
idem for these two

Reply:
Yes, I'll change the names. I think I'll go for `eq_singleton_[zero/one]_of_unique` for the ones using [Unique] and just `eq_singleton_[zero/one]` for the ones using `Subsingleton` since they are probably the most useful.